gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/DatastoreFactory.java

```diff
@@ -16,18 +16,29 @@
 
 package com.google.gcloud.datastore;
 
+import com.google.gcloud.ServiceFactory;
+import java.io.ObjectStreamException;
 
 /**
  * A base class for Datastore factories.
  */
-public abstract class DatastoreFactory {
+public abstract class DatastoreFactory implements ServiceFactory<Datastore, DatastoreOptions> {
+
+  private static final long serialVersionUID = 5037190305022535983L;
 
   private static final DatastoreFactory INSTANCE = new DatastoreFactory() {
```

Comment:
@eamonnmcmanus I think that Arie refers to the discussion we had in #55 following [this comment](https://github.com/GoogleCloudPlatform/gcloud-java/issues/55#issuecomment-141541987).